### ISSUE TITLE
bug: `label` not set in`metrics.summarize` when setting `pos_label` at the level of report

### ISSUE THREAD

**Issue by glemaitre** [probabl-ai/skore#2799]
### What would you like to say?

Given the following example:

```python
# %%
from sklearn.datasets import load_breast_cancer

breast_cancer = load_breast_cancer(as_frame=True)
X, y = breast_cancer.data, breast_cancer.target
y = breast_cancer.target_names[y]

# %%
import skrub
from sklearn.linear_model import LogisticRegression

estimator = skrub.tabular_pipeline(LogisticRegression())
estimator

# %%
import skore

report = skore.evaluate(estimator, X, y, splitter=0.25, pos_label="malignant")
report

# %%
report.metrics.summarize().frame()
```

The output will be:

metric | estimator_name | data_source | favorability | label | average | output | score
-- | -- | -- | -- | -- | -- | -- | --
Accuracy | LogisticRegression | test | (↗︎) | None | None | None | 0.965035
Precision | LogisticRegression | test | (↗︎) | None | None | None | 0.944444
Recall | LogisticRegression | test | (↗︎) | None | None | None | 0.962264
ROC AUC | LogisticRegression | test | (↗︎) | None | None | None | 0.994759
Log loss | LogisticRegression | test | (↘︎) | None | None | None | 0.085966
Brier score | LogisticRegression | test | (↘︎) | None | None | None | 0.026100
Fit time (s) | LogisticRegression | test | (↘︎) | None | None | None | 0.091657
Predict time (s) | LogisticRegression | test | (↘︎) | None | None | None | 0.021935

Basically, we don't set the `label` in this case. I think that we should. It is even inconsistent if we add a specific metric:

```python
# %%
report.metrics.add("f1")
report.metrics.summarize().data
```

that give:

  | metric | estimator_name | data_source | favorability | label | average | output | score
-- | -- | -- | -- | -- | -- | -- | -- | --
Accuracy | LogisticRegression | test | (↗︎) | None | None | None | 0.965035
Precision | LogisticRegression | test | (↗︎) | None | None | None | 0.944444
Recall | LogisticRegression | test | (↗︎) | None | None | None | 0.962264
ROC AUC | LogisticRegression | test | (↗︎) | None | None | None | 0.994759
Log loss | LogisticRegression | test | (↘︎) | None | None | None | 0.085966
Brier score | LogisticRegression | test | (↘︎) | None | None | None | 0.026100
Fit time (s) | LogisticRegression | test | (↘︎) | None | None | None | 0.091657
Predict time (s) | LogisticRegression | test | (↘︎) | None | None | None | 0.021935
F1 | LogisticRegression | test | (↗︎) | malignant | None | None | 0.953271

Where we see that the f1 score properly set the label in this case.

**Comment by auguste-probabl** [comment]
The `favorability` column has `"malignant"`, is it a typo?
